src/main/telemetry/hott.c

```diff
@@ -453,6 +530,565 @@ static void processBinaryModeRequest(uint8_t address)
 #endif
 }
 
+static void processTextModeRequest(HOTT_TEXTMODE_MSG_t *hottTEXTMODEMessage,uint8_t octet3)
+{
+	uint8_t id_sensor = (octet3 >> 4);
+	uint8_t id_key = octet3 & 0x0f;
+	static uint8_t ligne_select = 4 ;
+	static int8_t ligne_edit = -1 ;
+	char lineText[21];
+
+	hottTEXTMODEMessage->start_byte = 0x7b;
+	hottTEXTMODEMessage->esc = 0x00;
+	hottTEXTMODEMessage->warning_beeps = 0x00;
+	memset((char *)&hottTEXTMODEMessage->text, 0x20, HOTT_TEXTMODE_MSG_TEXT_LEN);
+	hottTEXTMODEMessage->stop_byte = 0x7D;
+	
+	if (id_key == HOTTV4_BUTTON_LEFT && page_settings == 1)
+        {   
+          hottTEXTMODEMessage->esc = 0x01;
+        }
+        else
+        {
+          if (id_sensor == (HOTT_TELEMETRY_EAM_SENSOR_ID & 0x0f) && (!ARMING_FLAG(ARMED))) // Only Text modus when "No ARMED"
+            {  
+            hottTEXTMODEMessage->esc = HOTT_TELEMETRY_EAM_SENSOR_TEXT;
+			
+			    // Button Abfrage 
+                if (ligne_edit == -1) //keine line gewählt
+                {            
+                    switch (id_key) {
+						case HOTTV4_BUTTON_LEFT:
+							{
+							page_settings-=1;
+							if (page_settings <1)    // unter Seite 1 dann Seite 2
+							page_settings = 1;
+							break;	
+							}
+						case HOTTV4_BUTTON_RIGHT:
+							{	   
+							page_settings+=1;
+							if (page_settings >8)   // Über Seite 8 dann Seite 1
+								page_settings = 1;
+							break;	
+						}
+						case HOTTV4_BUTTON_UP: 
+							{   
+							ligne_select+=1;
+							if (ligne_select>7)
+								ligne_select =  1;
+							break;
+						}
+						case HOTTV4_BUTTON_DOWN:
+							{   
+							ligne_select-=1;
+							if (ligne_select<1)
+								ligne_select =  1;
+							break;
+							}
+						case HOTTV4_BUTTON_SET:
+							{   
+							ligne_edit =  ligne_select ;
+							break;
+							}  
+					}			
+                }    
+                else if (ligne_edit != -1)
+                {
+					switch (id_key) {
+						case HOTTV4_BUTTON_RIGHT:
+							{
```

Comment:
No blocks in switch statements were not needed. Contract cases with identical implementation please.